org.avatar.gics.service/src/org/avatar/gics/service/impl/GICSServiceImpl.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2012 - 2024 Data In Motion and others.
+ * All rights reserved. 
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *     Data In Motion - initial API and implementation
+ */
+package org.avatar.gics.service.impl;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import org.avatar.gics.service.api.GICSService;
+import org.avatar.gics.service.api.GICSServiceConfig;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.util.FeatureMap;
+import org.emau.icmvc.ganimed.ttp.cm2.AddConsent;
+import org.emau.icmvc.ganimed.ttp.cm2.AddConsentResponse;
+import org.emau.icmvc.ganimed.ttp.cm2.Cm2Factory;
+import org.emau.icmvc.ganimed.ttp.cm2.Cm2Package;
+import org.emau.icmvc.ganimed.ttp.cm2.ConsentDTO;
+import org.emau.icmvc.ganimed.ttp.cm2.GetAllConsentedIdsFor;
+import org.emau.icmvc.ganimed.ttp.cm2.GetAllConsentedIdsForResponse;
+import org.gecko.emf.osgi.annotation.require.RequireEMF;
+import org.gecko.emf.osgi.constants.EMFNamespaces;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.xmlsoap.schemas.envelope.Body;
+import org.xmlsoap.schemas.envelope.DocumentRoot;
+import org.xmlsoap.schemas.envelope.Envelope;
+import org.xmlsoap.schemas.envelope.EnvelopeFactory;
+import org.xmlsoap.schemas.envelope.EnvelopePackage;
+
+/**
+ * 
+ * @author ilenia
+ * @since Nov 28, 2024
+ */
+@RequireEMF
+@Component(name = "GICSService", configurationPid = "GICSService", configurationPolicy = ConfigurationPolicy.REQUIRE)
+public class GICSServiceImpl implements GICSService {
+
+	@Reference(target = "(&("+EMFNamespaces. EMF_MODEL_CONTENT_TYPE+"=soap)("+EMFNamespaces.EMF_MODEL_NAME+"=cm2))")
```

Comment:
This targ binding is not needed, because you inject the cm2 package. therefore you can expect that this packge is registered. Because we register the generated EPackages after all EMF registration is done